tests/test_dependencies.py

```diff
@@ -471,37 +478,59 @@ def test_middleware_response_without_content_type(self):
 
-    def test_middleware_response_with_components_with_slash_dash_and_underscore(
-        self,
-    ):
+    def test_middleware_response_with_components_with_slash_dash_and_underscore(self):
         registry.register("dynamic", DynamicComponent)
+        registry.register("test-component", component=SimpleComponent)
+        registry.register("test/component", component=SimpleComponent)
+        registry.register("test_component", component=SimpleComponent)
 
-        component_names = [
-            "test-component",
-            "test/component",
-            "test_component",
-        ]
-        for component_name in component_names:
-            registry.register(name=component_name, component=SimpleComponent)
-            template_str: types.django_html = """
-                {% load component_tags %}
-                {% component_css_dependencies %}
-                {% component_js_dependencies %}
-                {% component "dynamic" is=component_name variable='value' / %}
-            """
-            template = Template(template_str)
-            rendered = create_and_process_template_response(
-                template, context=Context({"component_name": component_name})
-            )
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component_css_dependencies %}
+            {% component_js_dependencies %}
+            {% component "dynamic" is=component_name variable='value' / %}
+        """
+        template = Template(template_str)
 
+        def assert_dependencies(content: str):
             # Dependency manager script (empty)
-            self.assertInHTML('<script src="django_components/django_components.min.js"></script>', rendered, count=1)
+            self.assertInHTML('<script src="django_components/django_components.min.js"></script>', content, count=1)
 
             # Inlined JS
-            self.assertInHTML('<script>console.log("xyz");</script>', rendered, count=1)
+            self.assertInHTML('<script>console.log("xyz");</script>', content, count=1)
             # Inlined CSS
-            self.assertInHTML("<style>.xyz { color: red; }</style>", rendered, count=1)
+            self.assertInHTML("<style>.xyz { color: red; }</style>", content, count=1)
             # Media.css
-            self.assertInHTML('<link href="style.css" media="all" rel="stylesheet">', rendered, count=1)
+            self.assertInHTML('<link href="style.css" media="all" rel="stylesheet">', content, count=1)
 
-            self.assertEqual(rendered.count("Variable: <strong>value</strong>"), 1)
+        rendered1 = create_and_process_template_response(
+            template,
+            context=Context({"component_name": "test-component"}),
+        )
+
+        assert_dependencies(rendered1)
+        self.assertEqual(
+            rendered1.count('Variable: <strong data-djc-id-a1bc41 data-djc-id-a1bc42="">value</strong>'),
+            1,
```

Comment:
There's a couple of patterns to unpack in

    rendered1.count('Variable: <strong data-djc-id-a1bc41 data-djc-id-a1bc42="">value</strong>')

1. Where possible I used `self.assertInHTML` or `self.assertHTMLEqual`, but I think the latter doesn't work when I want to search for a combination of text + HTML, e.g. like how above I searched for `Variable: <strong>...`.

   So in those cases I opted for `self.assertEqual(str.count('Variable: <strong>...), 1)`.

   But when using `str.count()` or `self.assertEqual(str)`, I have to put in there the exact same string.

   On the other hand, `self.assertHTMLEqual()` includes some nicities like:
   - HTML attributes can be in any order
   - Whitespace between elements is ignored
   - Boolean HTML attributes like `data-djc-id-a1bc42=""` and `data-djc-id-a1bc42` are considered as identical

   So, because I used `str.count()`, I had to write extact string, so I had to write `data-djc-id-a1bc42=""` instead of simply `data-djc-id-a1bc42`.

2. As for why there are even both `data-djc-id-a1bc42=""` and `data-djc-id-a1bc42` formats present:

   When we use BS4 to process the HTML to insert the ID attributes, then what I think happens is that, the first time the boolean attribute like `data-djc-id-a1bc42` is inserted, it's inserted without the value part (without `=""`). But when HTML already contains boolean HTML attributes without values, then BS4 will re-format them into `data-djc-id-a1bc42=""`. Semantically they are the same.

3. And as for why both `data-djc-id-a1bc41` and `data-djc-id-a1bc42=""` (two different IDs) are on the same HTML element is because that `<strong>` is the root HTML element for two components, where one is nested in another, e.g. something like this:

   ```py
   @register("outer")
   class Outer(Component):
       template = """
         {% component "inner" / %}
       """
   
   @register("inner")
   class Inner(Component):
       template = """
          <strong>
            Hi!
          </strong>
       """
   ```

Reply:
Thanks, I should have been more explicit in my question. I was wondering about the id="" part. But this seems to be a quirk of the bs4 parser, check.